version: '3.7'

services:
  db:
    image: postgres:17-alpine
    container_name: ${COMPOSE_PROJECT_NAME}-db
    networks:
      - default
    volumes:
      - ./data/db:/var/lib/postgresql/data
    env_file:
      - .env
    restart: unless-stopped
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U listmonk']
      interval: 10s
      timeout: 5s
      retries: 6

  app:
    container_name: ${COMPOSE_PROJECT_NAME}-app
    image: listmonk/listmonk:latest
    labels:
      - traefik.enable=true
      - traefik.docker.network=${NETWORK}
      - traefik.http.routers.${COMPOSE_PROJECT_NAME}.rule=Host(${DOMAINS})
      - traefik.http.routers.${COMPOSE_PROJECT_NAME}.entrypoints=https
      - traefik.http.routers.${COMPOSE_PROJECT_NAME}.tls=true
      - traefik.http.routers.${COMPOSE_PROJECT_NAME}.tls.certresolver=letsencrypt
      - traefik.http.services.${COMPOSE_PROJECT_NAME}.loadbalancer.server.port=${PORT}
    networks:
      - default
    env_file:
      - .env
    depends_on:
      - db
    volumes:
      - ./data/app/config/config.toml:/listmonk/config.toml
    restart: unless-stopped
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/api/health"]
      interval: 10s
      timeout: 5s
      retries: 6

networks:
  default:
    external:
      name: ${COMPOSE_PROJECT_NAME}
